src/components/form/form-directive.ts

```diff
@@ -82,33 +83,49 @@ export class FormController {
             let schema = values[0];
             this.uiSchema = <IUISchemaElement> values[1];
             let data = values[2];
-
-            let dataProvider: IDataProvider;
-            let services = new Services();
-
-            services.add(new PathResolverService(new PathResolver()));
-            services.add(new ScopeProvider(this.scope));
-            services.add(new SchemaProvider(schema));
-            services.add(new ValidationService());
-            services.add(new RuleService(this.PathResolver));
-
-            if (FormController.isDataProvider(this.scope.data)) {
-                dataProvider = this.scope.data;
+            // this.render(schema, data);
+            let unresolvedRefs = JsonRefs.findRefs(schema);
+            if (unresolvedRefs === undefined || Object.keys(unresolvedRefs).length === 0) {
```

Comment:
Does `findRefs` ever return undefined? The length check could be replaced with `_.size`.